src/drt/src/io/io_parser_helper.cpp

```diff
@@ -919,46 +920,63 @@ void io::Parser::buildGCellPatterns_getOffset(frCoord GCELLGRIDX,
   }
 }
 
-void io::Parser::buildGCellPatterns()
+void io::Parser::buildGCellPatterns(odb::dbDatabase* db)
 {
   // horizontal = false is gcell lines along y direction (x-grid)
-  Rect dieBox;
-  design->getTopBlock()->getDieBox(dieBox);
-
+  frGCellPattern xgp, ygp;
   frCoord GCELLOFFSETX, GCELLOFFSETY, GCELLGRIDX, GCELLGRIDY;
-  buildGCellPatterns_helper(GCELLGRIDX, GCELLGRIDY, GCELLOFFSETX, GCELLOFFSETY);
+  auto gcellGrid = db->getChip()->getBlock()->getGCellGrid();
+  if (gcellGrid != nullptr && gcellGrid->getNumGridPatternsX() == 1
+      && gcellGrid->getNumGridPatternsY() == 1) {
```

Comment:
Why the restriction to one pattern?

Reply:
currently fastroute writes out 1 pattern only, so if it's not one pattern, it's not coming from fastroute. Also tritonroute use one pattern as well and I am not sure how to handle the case if I have more than one pattern, so I just leave it to tritonroute to figure out one pattern that satisfies the guides.

Reply:
ok